### DISCUSSION TITLE
Removing `any` types from `replace` and `merge` methods

### DISCUSSION THREAD

**Discussion by Nokel81** [mobxjs/mobx#3285] [Ideas]
The current typings for `ObservableMap<K, V>` include the following method:

```
replace(values: ObservableMap<K, V> | IKeyValueMap<V> | any): ObservableMap<K, V>;
```

I would love to see that `| any` removed, what could I do to help with that? Including for `ObservableSet` and `ObservableArray`.

I know that, for instance, also `[K, V][]` is an acceptable type for `values` since I use it a lot.

**Comment by kubk** [comment]
I've converted the discussion to an issue: https://github.com/mobxjs/mobx/issues/3286

Let's discussing it there 👍

**Reply by Nokel81** [reply]
Sounds, good I didn't know what to create
